alica_engine/include/engine/blackboard/Blackboard.h

```diff
@@ -333,12 +334,12 @@ class Blackboard
     std::shared_lock<std::shared_mutex> lockRO() const { return std::shared_lock(_mtx); }
     std::unique_lock<std::shared_mutex> lockRW() { return std::unique_lock(_mtx); }
 
-    // Not thread safe.  Avoid for public use
-    BlackboardImpl& impl() { return _impl; }
-    const BlackboardImpl& impl() const { return _impl; }
+    // TODO: once these APIs are removed from user code, make these methods private so that only alica internal's can use them
+    [[deprecated("Use UnlockedBlackboard instead")]] internal::BlackboardImpl& impl() { return _impl; }
```

Comment:
Build log will be overfilled with this warnings, correct me if I wrong.

Reply:
@veerajsk Please add a new method (private, not deprecated) that we use when constructing unlocked blackboard so that we don't get compiler warnings from normal use.  Currently even building alica will raise warnings because this function is used internally

Reply:
Or internal things just use _impl directly so we can remove this function entirely instead of making it private

Reply:
Idea was to use UnlockedBlackboard eveywhere & deprecate this method until then. But yes, this is generating a bunch of warnings even in the engine. I'll work on a fix

Reply:
I switched to using _impl directly for alica internals & tests. Had to declare a bunch of friends for it though. However, seems to be the best solution

Reply:
Maybe tests can just lock the blackboard?  Performance doesn't matter that much for tests and Locking really isn't that important to avoid anyway imo

Reply:
Still the motivation is unclear for me for this change. Seems code became worse after this change

Reply:
@Dale-Koenig the access to impl is only required for accessing the `map()` method, which is not exposed to the user. Otherwise the tests are mostly using the Locked blackboard methods

@ntfshard some context here: User code was using the impl() method to access the blackboard without a lock, which is bad, since user code should not be accessing internal implementation classes. However, as explained here https://github.com/rapyuta-robotics/alica/pull/425#issuecomment-1436461189 we need unlocked access to blackboard in many cases. Therefore this PR adds a UnlockedBlackboard class which the user code will now use instead of the impl() method. Once user code is change this deprecated method will be removed. I hope it is clear now?